### DISCUSSION TITLE
Themenfindung

### DISCUSSION THREAD

**Discussion by felixhoffmnn** [felixhoffmnn/aswe#1] [General]
## Beispiel Use Cases

1. **Fortbewegungsmittel** anhand von Rapla, Spracherkennung und Wetterdaten vorschlagen
   - Vorschläge für Fortbewegungsmittel: Zug oder Fahrrad
2. **Morning Briefing**
    - „Dashboard“ für den **Aktienmarkt**
      - Gibt Kursübersicht über die vergangenen Tage
    - **Tageszusammenfassung** (100 Tagesschau)
      - News, Wetter, Verkehrsdaten, Aktienmarkt, Kryptomarkt, Sport

**Comment by nbs2904** [comment]
## Quellen

- [Public APIs](https://github.com/public-apis/public-apis)
- [Liste der Deutschen Bahn APIs](https://data.deutschebahn.com/dataset.groups.apis.html)
- [Fußball Ergebnisse API](https://www.openligadb.de/)
- [Eventbrite](https://www.eventbrite.com/platform/api/#/introduction/about-our-api)

**Reply by felixhoffmnn** [reply]
- [VVS](https://github.com/zaanposni/vvspy)

**Comment by l7f19** [comment]
## Event Suggestion

- Terminfindung und Erstellung von Kalendereinträgen
  - Connection zu z.B. Wetter ("Wann nächster freier Abend mit gutem Wetter?") oder Events (z.B. "Kommende Konzerte/Veranstaltungen, zu denen ich Zeit habe?")

**Reply by felixhoffmnn** [reply]
Weitere Fragebeispiele:

- Was könnte ich heute Abend machen?

Für die Vorschläge kann der Nutzer einerseits eine **Abfrage am Anfang** durchführen. Außerdem könnte die API auf Labels von **Terminen aus dem Kalender** zugreifen. Zusammenfassend würde die API basierend auf der _Frage_ (also dem Kontext), _historischen Terminen_, _Wetter_ und _lokalen Veranstaltungen_ Termine und Co. vorschlagen. Neben der Spracheingabe könne die API auch als Widget in Kalenderanwendungen eingebettet werden.

**Reply by nbs2904** [reply]
Für die Events könnten wir eventuell die API von [Eventbrite](https://www.eventbrite.com/platform/api/#/introduction/about-our-api) nutzen, sofern diese ausreicht. 
Vielleicht, dass wir auch so eine allgemeine "intelligente" ToDo Applikation entwickeln, in der man alles simple hinzufügt und die Applikation versucht den besten Timeslot für das ToDo / Event anhand von Wetter / Eventlänge / anderen Terminen, etc... zu finden.

**Comment by steadl** [comment]
## Irgendwas + Übersetzer

- Messenger + Übersetzer
    - Empfang Nachricht in Englisch, Übersetze in Deutsch + Sprachausgabe
    - Empfange Nachricht in beliebiger Sprache, Sprachidentifikation + Übersetzung
- Entertainment API + Übersetzer
    - zufälliger englischer Witz wird auf deutsch übersetzt und ausgegeben
    - Excuser API + Übersetzer (Ausrede für gesagte Situation auf Englisch, ins Deutsche Übersetzt)
    - Fun Fact + Übersetzer

**Comment by steadl** [comment]
Wir können auch irgendeine Art von (Preis-)Vergleich mit mehreren APIs machen, damit wir auf 4 verschiedene APIs kommen

**Comment by nbs2904** [comment]
Rezept / Getränke Vorschläge anhand von vorhandenen Lebensmitteln
- https://rapidapi.com/apidojo/api/tasty/

**Comment by mFritzsch** [comment]
### **Sport Tool**
Man gibt per Spracheingabe eine Zeitdauer an, die man joggen oder radfahren will. Dann

- Wird Strava oder Fitbit oder eine ähnliche API genutzt, um die durchnittliche Geschwindigkeit oder so etwas herauszufinden, mit welcher man dann potentiell die Routenlänge berechnen kann
- Wird eine Maps API genutzt, um eine Route zu erzeugen
- Wird eine Musik API genutzt, um für diese Dauer eine Playlist zu erstellen
- Wird die Wetter API genutzt, um einen guten Zeitpunkt für die Sporteinheit vorzuschlagen (oder schlägt Kleidung vor)
